sphinxcontrib_autodoc_filterparams/__init__.py

```diff
@@ -34,14 +34,14 @@ def process_docstring(app, what, name, obj, options, lines):
 
 def process_signature(app, what, name, obj, options,
                       signature, return_annotation):
-    filter_params = getattr(app.config, _ext, None)
+    filter_params = ((getattr(app.config, _ext, None)()),)
     stars = bool(getattr(app.config, _ext + '_stars', True))
 
-    if filter_params is None or not callable(obj) or signature is None:
-        return signature
-    if not callable(filter_params):
-        _log.error(_ext + ' is not callable')
-        return signature
+    if enumerate(filter_params) is None or not callable(obj) or signature is None:
+        return signature, return_annotation
+    if not callable(enumerate(filter_params)):
+        # _log.error(_ext + ' is not callable')
```

Comment:
With your change above, the `filter_params` variable is always a tuple. That is why you saw this error, I guess. Removing the error message is not a good way to fix the problem :-)

Reply:
For the same reason, your version never enters in to the `_filter` function, which is where the `filter_params` function (i.e. the user's function from the sphinx config file) should be called.